lib/preload.js

```diff
@@ -494,7 +494,31 @@ class Preloader extends EventEmitter {
       return Promise.resolve()
     }
     return this._runWithSpinner(`Fetching application ${this.appId}`, () => {
-      return this.balena.models.application.get(this.appId, {$expand: applicationExpandOptions})
+      const releaseFilter = {
+        status: 'success'
+      }
+      if (this.commit != null) {
+        releaseFilter.commit = { $startswith: this.commit }
+      }
+      return this.balena.models.application.get(this.appId, {
+        $expand: {
+          owns__release: {
+            $select: ['id', 'commit', 'end_timestamp', 'composition'],
+            $orderby: [{ end_timestamp: 'desc' }, { id: 'desc' }],
+            $expand: {
+              contains__image: {
+                $select: ['image'],
+                $expand: {
+                  image: {
+                    $select: ['image_size', 'is_stored_at__image_location']
+                  }
+                }
+              }
+            },
+            $filter: releaseFilter
+          }
+        }
+      })
```

Comment:
@Page- can't we reuse `preload.applicationExpandOptions` here?

Something like
```javascript
const options = _.cloneDeep(preload.applicationExpandOptions);
if (this.commit != null) {
    options.owns__release.$filter.commit = { $startswith: this.commit }
}
return this.balena.models.application.get(this.appId, {$expand: options})
```

Reply:
So the plan was to move more of the querying/filtering into this directly, but because of the way all the usage is spread out I haven't managed to move more filters in by eye - but this one alone gives us a workaround for the issue for now at least